pahfit/features/features.py

```diff
@@ -0,0 +1,327 @@
+                                         f" format at {name}\n\t{file}")
+
+            try: kind = elem.pop('kind')
+            except KeyError:
+                raise PAHFITFeatureError(f"No kind found for {name}\n\t{file}")
+            
+            try: valid_params = cls._kind_params[kind]
+            except KeyError:
+                raise PAHFITFeatureError(f"Unknown kind {kind} for {name}\n\t{file}")
+            unknown_params = [x for x in keys
+                              if not (x in valid_params or x in cls._group_attrs)]
+            if unknown_params:
+                raise PAHFITFeatureError(f"Unknown {kind} parameters:"
+                                         f" {', '.join(unknown_params)}\n\t{file}")
+
+            hasFeatures = 'features' in elem
+            hasLists = any(k not in cls._group_attrs and
+                           isinstance(v, (tuple,list,dict))
+                           for (k,v) in elem.items())
+            if hasFeatures and hasLists:
+                raise PAHFITFeatureError("A single group cannot contain both 'features'"
+                                         f" and parameter list(s): {name}\n\t{file}")
+            isGroup = (hasFeatures or hasLists)
+            bounds = None
+            if isGroup: # A named group of features
+                if 'bounds' in elem:
+                    if not isinstance(elem['bounds'], dict):
+                        for p in cls._no_bounds:
+                            if p in elem:
+                                raise PAHFITFeatureError(f"Parameter {p} cannot have "
+                                                         f"bounds: {name}\n\t{file}")
+                        if sum(p in elem for p in valid_params) > 1:
+                            raise PAHFITFeatureError("Groups with simple bounds "
+                                                     "can only specify a single "
+                                                     f"parameter: {name}\n\t{file}")
+                        if hasFeatures:
+                            raise PAHFITFeatureError("Groups with simple bounds "
+                                                     "cannot specify "
+                                                     f"'features': {name}\n\t{file}")
+                    bounds = elem.pop('bounds')
+                if hasFeatures: # our group uses a features: key
+                    for (n,v) in elem['features'].items():
+                        if bounds and not 'bounds' in v: # inherit bounds
+                            v['bounds'] = bounds
+                        cls._add_feature(kind, feat_tables, n, group=name, **v)
+                elif hasLists:
+                    llen = []
+                    for k,v in elem.items():
+                        if k in cls._group_attrs: continue
+                        if not isinstance(v, (tuple,list,dict)):
+                            raise PAHFITFeatureError(f"All non-group parameters in {name} "
```

Comment:
This part of the code raised the error.

Reply:
Thanks @Ameek-Sidhu, I just uploaded a fix for this; can you try it out?